### PR TITLE
Stop Two-way sync of gateway models and remove couch gateway models

### PR DESCRIPTION
**Pull request by gcapalbo**
@esoergel once https://github.com/dimagi/commcare-hq/pull/10057 is merged, I'll merge this with master to reduce the diff

### CODE REVIEW

File: corehq/apps/sms/util.py [hunk 1]
```diff
@@ -231,8 +215,24 @@ def get_backend_name(backend_id):
         return None
 
     try:
-        doc = MobileBackend.get_db().get(backend_id)
-    except ResourceNotFound:
+        return SQLMobileBackend.load(backend_id, is_couch_id=True).name
```
Comment (dimagimon) [inline corehq/apps/sms/util.py]: F821 undefined name 'SQLMobileBackend'

File: corehq/apps/sms/views.py [hunk 2]
```diff
@@ -40,17 +42,18 @@
 from corehq.apps.users.views.mobile.users import EditCommCareUserView
 from corehq.apps.sms.models import (
     SMSLog, INCOMING, OUTGOING, ForwardingRule,
-    LastReadMessage, MessagingEvent, SelfRegistrationInvitation
+    LastReadMessage, MessagingEvent, SelfRegistrationInvitation,
+    SQLMobileBackend, SQLMobileBackendMapping, PhoneLoadBalancingMixin
 )
-from corehq.apps.sms.mixin import (SMSBackend, BackendMapping, VerifiedNumber,
-    SMSLoadBalancingMixin, UnrecognizedBackendException)
+from corehq.apps.sms.mixin import (VerifiedNumber,
```
Comment (dimagimon) [inline corehq/apps/sms/views.py]: F401 'UnrecognizedBackendException' imported but unused

File: corehq/messaging/smsbackends/tropo/views.py [hunk 3]
```diff
@@ -68,7 +68,7 @@ def ivr_in(request):
             phone_number = cleaned_number,
             direction = INCOMING,
             date = datetime.utcnow(),
-            backend_api = TropoBackend.get_api_id(),
+            backend_api = SQLTropoBackend.get_api_id(),
```
Comment (dimagimon) [inline corehq/messaging/smsbackends/tropo/views.py]: E251 unexpected spaces around keyword / parameter equals

**Comment by gcapalbo** [comment]
@esoergel just merged master so the diff is much easier, this one is mostly deletions

File: corehq/apps/sms/models.py [hunk 4]
```diff
@@ -1455,7 +1455,7 @@ def get_queryset(self):
         return super(ActiveMobileBackendManager, self).get_queryset().filter(deleted=False)
 
 
-class SQLMobileBackend(SyncSQLToCouchMixin, models.Model):
+class SQLMobileBackend(models.Model):
```
Comment (esoergel) [inline corehq/apps/sms/models.py]: :fireworks:

**Comment by czue** [comment]
Good numbers 
image

**Pull request merged**